File: LeetCode/Easy/0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    prev := nums[0];
    cnt := 0;
    maxCnt := 0;
    majority := nums[0];
    slices.Sort(nums);
    
    for i := 0; i < len(nums); i++ {
        if (nums[i] == prev) {
            cnt += 1;
        } else {
            prev = nums[i];
            cnt = 1;
        }

        if (maxCnt < cnt ) {
            maxCnt = cnt;
            majority = prev;
        }
    }
    if (maxCnt < cnt ) {
        maxCnt = cnt;
        majority = prev;
    }
    return majority;
}

File: LeetCode/Easy/0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int m = (int)(n/2+1);

        Arrays.sort(nums);

        int x = nums[0];
        int cnt = 0;
        // System.out.println(m);
        for (int i = 0; i < n; i++) {
            // System.out.println(x +", " + cnt);
            if (x == nums[i]) {
                cnt ++;
            }
            if (cnt == m) {
                break;
            }
            if (x != nums[i]) {
                x = nums[i];
                cnt = 1;
            }
        }
        
        return x;
    }
}